### PR TITLE
Setup jest test environment

### PR DESCRIPTION
**Pull request by 0xGabi**
Boostrap an initial testing environment using Jest 🃏setup with Typescript, using [`ts-jest`](https://kulshekhar.github.io/ts-jest/) library. Only set up tests for the core library and main connectors. 

The PR also add a GitHub action to setup Codecov. Note Jest already support coverage out of the box 🙌

I decided to include a couple of scripts to support a few "test modes":
1. `test:coverage`: Will run `jest --coverage`
2. `test:watch`: Will run `jest --watch` (super powerful Jest watch mode 🧙‍♂️)
3. `test`: Will run `is-ci "test:coverage" "test:watch"`
   - If we are in local development will run the watch script. 
   - If we are in CI will run the coverage script.
4. `test:debug`: Will run `node --inspect-brk node_modules/jest/bin/jest.js --runInBand --watch` this piece of logic will start a node process listening for a debugging client and will attach the Jest binary. If we place a `debugger` statement in our tests code we will be able to use the browser/IDE to debug the attached process!! 

We might also want to fix lint to include it as a required step of the CI.

This documentation might be relevant to better understand the context of the options on the `jest.config` file:
- [Configuring Jest](https://jestjs.io/docs/en/configuration#projects-arraystring--projectconfig)

---

Note that the convention I decided to use for tests was: use `__test__` folders really close to the logic that is tested across the different projects. Would like to know others opinion about it.

### CODE REVIEW

File: jest.config.base.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
```
Comment (0xGabi) [inline jest.config.base.js:2]: Here is where we tell Jest to use the typescript library `ts-jest`.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const baseConfig = require('./jest.config.base')
```
Comment (0xGabi) [inline jest.config.js:1]: This file is where we choose the global jest configurations.

File: package.json [hunk 3]
```diff
@@ -34,14 +34,22 @@
...
+    "test:debug": "node --inspect-brk node_modules/jest/bin/jest.js --runInBand --watch",
     "clean": "yarn clean && oao clean",
     "build": "MODE=libs-only ./scripts/rebuild-packages.sh",
     "build:examples": "MODE=all ./scripts/rebuild-packages.sh",
     "status": "oao status",
     "postinstall": "yarn build"
   },
   "devDependencies": {
-    "oao": "^1.8.0"
+    "@types/jest": "^26.0.0",
```
Comment (0xGabi) [inline package.json:48]: This package is required by `ts-jest`.

File: packages/connect-core/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const packageName = require('./package.json').name.split('@aragon/').pop()
+
+module.exports = {
+  ...baseConfig,
+  roots: [`<rootDir>/packages/${packageName}`],
+  collectCoverageFrom: ['src/**/*.{ts,tsx}'],
+  testRegex: `(packages/${packageName}/.*/__tests__/.*|\\.(test|spec))\\.tsx?$`,
+  modulePaths: [`<rootDir>/packages/${packageName}/src/`],
+  name: packageName,
+  displayName: 'CORE',
```
Comment (0xGabi) [inline packages/connect-core/jest.config.js:12]: All project-specific files are pretty similar, only the display name change.

File: packages/connect-core/package.json [hunk 5]
```diff
@@ -7,15 +7,23 @@
   "scripts": {
     "build": "yarn compile",
     "dev": "yarn build --watch",
+    "test": "is-ci \"test:coverage\" \"test:watch\"",
```
Comment (0xGabi) [inline packages/connect-core/package.json:10]: I decide to keep the same scripts on every project and in the global scope. Let me know what you think.

File: packages/connect-core/src/utils/__test__/address.test.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { addressesEqual, includesAddress, ANY_ENTITY } from '../address'
```
Comment (0xGabi) [inline packages/connect-core/src/utils/__test__/address.test.ts:1]: A first test example.

**Comment by ajsantander** [comment]
Soooo happy with this PR! Thank you @0xGabi

**Comment by sohkai** [comment]
@0xGabi 

> I decided to include a couple of scripts to support a few "test modes":

These are great! Let's leave a note in the README or CONTRIBUTING to note these helpful scripts down!

**Pull request merged**